video/out/vo_x11.c

```diff
@@ -102,16 +118,19 @@ static bool getMyXImage(struct priv *p, int foo)
             MP_WARN(vo, "Shared memory error,disabling ( Ximage error )\n");
             goto shmemerror;
         }
-        p->Shminfo[foo].shmid = shmget(IPC_PRIVATE,
-                                       p->myximage[foo]->bytes_per_line *
-                                       p->myximage[foo]->height,
-                                       IPC_CREAT | 0777);
+        len = p->myximage[foo]->bytes_per_line * p->myximage[foo]->height;
+        memcpy(p->shmname, "/tmp/mpv-x11-XXXXXXXXXX", sizeof(p->shmname));
```

Comment:
shouldn't this rely on `TMPDIR`?

Reply:
This should be `memfd_create` and on platforms that doesn't support it, it should be made as a platform specific impl in `osdep`